java/rocksjni/portal.h

```diff
@@ -2477,6 +2478,33 @@ class JniUtil {
     return op(key_slice);
   }
 
+  /*
+   * Helper for operations on a key on java byte array
+   * Copies values from jbyte array to slice and performs op on the slice.
+   * for example WriteBatch->Delete
+   *
+   * from `op` and used for RocksDB->Delete etc.
+   */
+  static void k_op_indirect(std::function<void(ROCKSDB_NAMESPACE::Slice&)> op,
+                            JNIEnv* env, jbyteArray jkey, jint jkey_off,
+                            jint jkey_len) {
+    if (jkey == nullptr || env->GetArrayLength(jkey) < (jkey_off + jkey_len)) {
+      ROCKSDB_NAMESPACE::RocksDBExceptionJni::ThrowNew(env,
+                                                       "Invalid key argument");
+      return;
+    }
+    const std::unique_ptr<char[]> target(new char[jkey_len]);
+    if (target == nullptr) {
+      ROCKSDB_NAMESPACE::OutOfMemoryErrorJni::ThrowNew(env,
+           "Memory allocation failed in RocksDB JNI function");
+      return;
+    }
+    env->GetByteArrayRegion(jkey, jkey_off, jkey_len,
+                            reinterpret_cast<jbyte*>(target.get()));
+    ROCKSDB_NAMESPACE::Slice target_slice(target.get(), jkey_len);
```

Comment:
Missing error check on the JNI call before this, e.g. `if (env->ExceptionOccurred())`